src/main/groovy/com/marklogic/gradle/task/roxy/RoxyGenerateTestScaffoldTask.groovy

```diff
@@ -0,0 +1,100 @@
+        def suiteSetupXqyString = SampleTestFiles.getSuiteSetupXqy()
+        processTemplateString(engine, binding, arguments.targetSuiteDirName, "/suite-setup.xqy", suiteSetupXqyString)
+        def suiteTeardownXqyString = SampleTestFiles.getSuiteTeardownXqy()
+        processTemplateString(engine, binding, arguments.targetSuiteDirName, "/suite-teardown.xqy", suiteTeardownXqyString)
+        def setupXqyString = SampleTestFiles.getSetupXqy()
+        processTemplateString(engine, binding, arguments.targetSuiteDirName, "/setup.xqy", setupXqyString)
+
+        def sourceTestDataDirectory = project.file(DefaultValues.resourcesDirName + "/" + DefaultValues.sampleDirName + "/test-data")
+        def targetTestDataDirectoryName = arguments.targetSuiteDirName + "/test-data"
+        def targetTestDataDirectory = project.file(targetTestDataDirectoryName)
+        project.file(targetTestDataDirectoryName).mkdirs()
+        project.copy {
+            from sourceTestDataDirectory
+            into targetTestDataDirectory
+            include '**'
+        }
+    }
+
+    def processTemplateFile(engine, binding, targetBaseDir, templateFilePath) {
+        def templateFile = project.file(DefaultValues.resourcesDirName + "/" + DefaultValues.templateDirName + templateFilePath)
+        def template = engine.createTemplate(templateFile).make(binding)
+        def templateResult = template.toString()
+        project.file(targetBaseDir + templateFilePath).write(templateResult)
+    }
+
+    def processTemplateString(engine, binding, targetBaseDir, templateFilePath, templateString) {
+        def template = engine.createTemplate(templateString).make(binding)
+        def templateResult = template.toString()
+        project.file(targetBaseDir + templateFilePath).write(templateResult)
+    }
+
+    class DefaultValues {
+        static String defaultRoxySuitesDirName = 'roxy/src/test/suites'
+        static String defaultSuiteName = 'SampleTestSuite'
+        static String boilerplateSuiteName = 'BoilerplateTestSuite'
+        static String templateDirName = 'test-templates'
+        static String defaultTestName = 'SampleTest'
+        static String resourcesDirName = 'src/main/resources'
+        static String sampleDirName = 'sample-tests'
+        static String boilerplateDirName = 'boilerplate-tests'
+        static String defaultTargetSuiteDirName = defaultRoxySuitesDirName + '/' + defaultSuiteName
+        static String boilerplateSuiteDirName = defaultRoxySuitesDirName + '/' + boilerplateSuiteName
+    }
+
+    class CommandLineArguments {
+        String roxySuitesDirName = DefaultValues.defaultRoxySuitesDirName
+        String suiteName = DefaultValues.defaultSuiteName
+        String targetSuiteDirName = roxySuitesDirName + '/' + suiteName
+        String testName = DefaultValues.defaultTestName
+
+        CommandLineArguments() {
+            if (project.hasProperty('roxySuitesDir')) {
```

Comment:
I think this can be removed, as noted above - just use getAppConfig().getModulePaths. An end user can then set mlModulePaths to define where the test is generated.